front/components/vaults/CreateOrEditVaultModal.tsx

```diff
@@ -312,6 +269,25 @@ export function CreateOrEditVaultModal({
               </div>
             )}
           </div>
+          {vault && (
+            <>
+              <Page.Separator />
+              <ConfirmDeleteVaultDialog
+                vault={vault}
+                handleDelete={onDelete}
+                isOpen={showDeleteConfirmDialog}
+                onClose={() => setShowDeleteConfirmDialog(false)}
+              />
+              <div className="flex w-full justify-end">
+                <Button
```

Comment:
We've discussed this broadly last week but we agreed to add the button in the modal, and the modal takes care of handling it's own state.

Reply:
let's revisit in another PR if you don't mind.